tests/unit/process_product_edit_rules.t

```diff
@@ -0,0 +1,215 @@
+	},
+	{
+		id => "block_by_category_tag_no_match",
+		desc => "Block for product with a category, does not block non matching",
+		edit_rules => [
+			{
+				name => "Block all but test",
+				conditions => [["in_category_tags", "en:test_cat"]],
+				actions => [["ignore"]]
+			},
+		],
+		product => {category_tags => ["en:other_cat"]},
+		result => 1,
+	},
+	{
+		id => "ignore_if_existing_ingredients_text_fr",
+		desc => "Remove edit on a ingredients text in french if one already exists",
+		edit_rules => [{name => "Disallow ingredients", actions => [["ignore_if_existing_ingredients_text_fr"]]},],
+		product => {ingredients_text_fr => "YES"},
+		form => {ingredients_text_fr => "NOPE"},
+		delete_param => ["ingredients_text_fr", "ingredients_text"],
+		result => 1,
+	},
+	{
+		id => "ignore_if_existing_ingredients_text_fr_no_lang_suffix",
+		desc => "Remove edit on a ingredients text in french if one already exists works without lang prefix",
+		edit_rules => [{name => "Disallow ingredients", actions => [["ignore_if_existing_ingredients_text_fr"]]},],
+		product => {ingredients_text_fr => "YES"},
+		form => {ingredients_text => "NOPE"},
+		delete_param => ["ingredients_text_fr", "ingredients_text"],
+		result => 1,
+	},
+	{
+		id => "ignore_if_existing_ingredients_text_fr_empty",
+		desc =>
+			"Remove edit on a ingredients text in french if one already exists not applicable because there is none",
+		edit_rules => [{name => "Disallow ingredients", actions => [["ignore_if_existing_ingredients_text_fr"]]},],
+		form => {ingredients_text_fr => "NOPE"},
+		result => 1,
+	},
+	{
+		id => "ignore_if_0_nutriment_fruits_vegetables_nuts",
+		desc => "Remove edit if 0 value on a nutriment field",
+		edit_rules =>
+			[{name => "Disallow ingredients", actions => [["ignore_if_0_nutriment_fruits-vegetables-nuts"]]},],
+		form => {"nutriment_fruits-vegetables-nuts" => 0},
+		result => 1,
+		delete_param => ["nutriment_fruits-vegetables-nuts"],
+	},
+	{
+		id => "ignore_if_equal_nutriments_sugar",
+		desc => "Remove edit if value on a nutriment field is equal to a value",
+		edit_rules => [{name => "Disallow ingredients", actions => [["ignore_if_equal_nutriment_sugar", 100]]},],
+		product => {"nutriment" => {"sugar_100g" => 22}},
+		form => {"nutriment_sugar" => 100},
+		result => 1,
+		delete_param => ["nutriment_sugar"],
+	},
+	{
+		id => "ignore_if_match_serving_size",
+		desc => "Remove edit if string value match a value",
+		edit_rules => [{name => "Disallow ingredients", actions => [["ignore_if_match_serving_size", "serving"]]},],
+		form => {"serving_size" => "serving"},
+		result => 1,
+		delete_param => ["serving_size"],
+	},
+	{
+		id => "ignore_if_regexp_match_brand",
+		desc => "Remove edit if string value match a regexp",
+		edit_rules =>
+			[{name => "Disallow ingredients", actions => [["ignore_if_regexp_match_brands", "(acme|hacky)"]]},],
+		form => {"brands" => "Another, Acme inc."},
+		result => 1,
+		delete_param => ["brands"],
```

Comment:
Maybe it would be clearer to rename "result" to "expected_result" and "delete_param" to "expected_deleted_params".

Reply:
I prefer to keep names sleek, maybe I can document the model above.